### PR TITLE
Add findBugs asserts

### PR DESCRIPTION
**Pull request by ikatkov**

### CODE REVIEW

**Comment by ikatkov** [comment]
Rebased

File: findbugs-exclude.xml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+<FindBugsFilter>
```
Comment (francisco-perez-sorrosal) [inline findbugs-exclude.xml]: Move this file to the misc/ directory

File: pom.xml [hunk 2]
```diff
@@ -304,6 +304,26 @@
                     <version>${maven-coveralls-plugin.version}</version>
                 </plugin>
 
+
+                <plugin>
+                    <groupId>org.codehaus.mojo</groupId>
+                    <artifactId>findbugs-maven-plugin</artifactId>
+                    <version>3.0.1</version>
```
Comment (francisco-perez-sorrosal) [inline pom.xml]: Create property for version:
        <findbugs-maven-plugin>3.0.1</findbugs-maven-plugin>
and add the version here:
        <version>${findbugs-maven-plugin.version}</version>

File: pom.xml [hunk 3]
```diff
@@ -304,6 +304,26 @@
...
+                    <configuration>
+                        <findbugsXmlOutput>true</findbugsXmlOutput>
+                        <xmlOutput>true</xmlOutput>
+                        <threshold>low</threshold>
+                        <excludeFilterFile>findbugs-exclude.xml</excludeFilterFile>
```
Comment (francisco-perez-sorrosal) [inline pom.xml]: Move the file to the misc/ directory:
`<excludeFilterFile>misc/findbugs-exclude.xml</excludeFilterFile>`

Comment (ikatkov) [inline pom.xml]: done

File: pom.xml (hunk 2 above)
Comment (ikatkov) [inline pom.xml]: done

File: findbugs-exclude.xml (hunk 1 above)
Comment (ikatkov) [inline findbugs-exclude.xml]: dup?

**Pull request merged**